owncloudApp/src/main/java/com/owncloud/android/presentation/ui/accounts/AccountManagementActivity.kt

```diff
@@ -0,0 +1,298 @@
+    private lateinit var tintedCheck: Drawable
+
+    private val removeAccountDialogViewModel: RemoveAccountDialogViewModel by viewModel()
+    private val accountsManagementViewModel: AccountsManagementViewModel by viewModel()
+
+    @Override
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.accounts_layout)
+        tintedCheck = ContextCompat.getDrawable(this, R.drawable.ic_current_white)!!
+        tintedCheck = DrawableCompat.wrap(tintedCheck)
+        val tint = ContextCompat.getColor(this, R.color.actionbar_start_color)
+        DrawableCompat.setTint(tintedCheck, tint)
+
+        val recyclerView: RecyclerView = findViewById(R.id.account_list_recycler_view)
+        recyclerView.run {
+            filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(applicationContext)
+            adapter = accountListAdapter
+            layoutManager = LinearLayoutManager(this@AccountManagementActivity)
+        }
+
+        setupStandardToolbar(
+            getString(R.string.prefs_manage_accounts),
+            displayHomeAsUpEnabled = true,
+            homeButtonEnabled = true,
+            displayShowTitleEnabled = true
+        )
+
+        val accountList = accountsManagementViewModel.getLoggedAccounts()
+        originalAccounts = toAccountNameSet(accountList)
+        originalCurrentAccount = accountsManagementViewModel.getCurrentAccount()?.name.toString()
+
+
+        accountListAdapter.submitAccountList(accountList = getAccountListItems())
+
+        account = accountsManagementViewModel.getCurrentAccount()
+        onAccountSet(false)
+
+    }
+
+    /**
+     * converts an array of accounts into a set of account names.
+     *
+     * @param accountList the account array
+     * @return set of account names
+     */
+    private fun toAccountNameSet(accountList: Array<Account>): Set<String> {
+        val actualAccounts: MutableSet<String> = HashSet(accountList.size)
+        for (account in accountList) {
+            actualAccounts.add(account.name)
+        }
+        return actualAccounts
+    }
+
+    override fun onBackPressed() {
+        val resultIntent = Intent()
+        resultIntent.putExtra(KEY_ACCOUNT_LIST_CHANGED, hasAccountListChanged())
+        resultIntent.putExtra(KEY_CURRENT_ACCOUNT_CHANGED, hasCurrentAccountChanged())
+        setResult(RESULT_OK, resultIntent)
+        finish()
+        super.onBackPressed()
+    }
+
+    /**
+     * checks the set of actual accounts against the set of original accounts when the activity has been started.
+     *
+     * @return `true` if account list has changed, `false` if not
+     */
+    private fun hasAccountListChanged(): Boolean {
+        val accountList = accountsManagementViewModel.getLoggedAccounts()
+        val actualAccounts = toAccountNameSet(accountList)
+        return originalAccounts != actualAccounts
```

Comment:
I'm not sure these can be compared with `==` or `!=` 🤔

Reply:
I'm not sure what you mean here and how to change it 😟  🤔

Reply:
I mean, I'm not sure if two String sets can be compared using `==` or `!=` operators, maybe this compares the reference and not the actual content of the arrays, causing always a `false` as result. Take a look at this.

Reply:
I've been testing few things and reading some documentation about it and if I change the comparison operator "==" for the function "equals()" android studio suggest changing it back to "==".
Also i read that "==" operator compares the content of the arrays and the referential equality operator is "===".

So it is working fine as it is done, but thanks for the warn so i could check it and learn some more 😄